Bugfix/714 catalog should give proper error message in posting

ywkim312 opened the pull request:
## Description
@sandeep-ps I found that Wenjie already made a page for error. I used it for now but I am not sure if it is okay for you. Also, I changed the message that appears when pushing the submit button but I am not also very sure if we need that message either.

**Fixes #(add issue number here and remove parentheses)**

[comment]: # (This project only accepts pull requests related to open issues.)
[comment]: # (If suggesting a new feature or change, please discuss it in an issue first.)
[comment]: # (If fixing a bug, there should be an issue describing it with steps to reproduce.)

## Review Time Estimate
_Please give your idea of how soon this pull request needs to be reviewed by selecting one of the options below. This can be based on the criticality of the issue at hand and/or other relevant factors._

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] Immediately
- [x] Within a week
- [ ] When possible

## Type of changes
_Please select a relevant option:_

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [x] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Other (any another change that does not fall in one of the above categories.)

## Checklist:
_Please select all applicable options:_

[comment]: # (To select your options, please put an 'x' in the all boxes that apply.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [x] I have signed the Rokwire Contributor License Agreement ([CLA](https://rokwire.org/rokwire_cla)). (_Any contributor who is not an employee of the University of Illinois whose official duties include contributing to the Rokwire software, or who is not paid by the Rokwire project, needs to sign the CLA before their contribution can be accepted._)
- [x] I have updated the [CHANGELOG](../CHANGELOG.md).
- [x] I have read the [Contributor Guidelines](../CONTRIBUTING.md).
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] My change requires updating the documentation.
- [ ] I have made necessary changes to the documentation.
- [ ] I have added tests related to my changes.
- [ ] My changes generate no new warnings.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

---

[comment]: # (Template credit: This pull request template is based on Embedded Artistry {https://github.com/embeddedartistry/templates/blob/master/.github/PULL_REQUEST_TEMPLATE.md}, Clowder {https://github.com/clowder-framework/clowder/blob/develop/.github/PULL_REQUEST_TEMPLATE.md}, and TalAter {https://github.com/TalAter/open-source-templates} templates.)

sandeep-ps commented:
@ywkim312, I think it's better to use Bootstrap alerts. Something that we can use to reuse on different pages. I tested this by shutting down the Contributions BB (to imitate backend failure), but Catalog is just displaying the stack trace on the page. Could you please check when you get a chance?

ywkim312 commented:
> @ywkim312, I think it's better to use Bootstrap alerts. Something that we can use to reuse on different pages. I tested this by shutting down the Contributions BB (to imitate backend failure), but Catalog is just displaying the stack trace on the page. Could you please check when you get a chance?

Okay will do

ywkim312 commented:
> @ywkim312, I think it's better to use Bootstrap alerts. Something that we can use to reuse on different pages. I tested this by shutting down the Contributions BB (to imitate backend failure), but Catalog is just displaying the stack trace on the page. Could you please check when you get a chance?

@sandeep-ps I found why it all shows the whole message. I changed it to give shorter message and you should be able to see it from the latest commit. It was because the code displaying whole error message on the error page. So I changed the code to log the whole error message and just give short error message in the error page. However, I am not very confident about the message content and the way I did. If you have better idea about the error message content, or way of handling this, please let me know.

sandeep-ps commented on contributions/catalog/controllers/contribute.py:
```diff
@@ -117,16 +118,17 @@ def create():
+                return render_template('contribute/submitted.html')
+        elif not response:
+            logging.error(s)
+            s = "There was a problem in POST process."
```
```suggestion
            s = "Contribution submission failed. Please try again after some time!"
```

sandeep-ps replied:
@ywkim312, please see my suggestion. Otherwise, this looks ready to be merged. Thanks.

ywkim312 replied:
changed.

sandeep-ps commented on CHANGELOG.md:
```diff
@@ -26,6 +26,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Fix title search with user auth key. [#732](https://github.com/rokwire/rokwire-building-blocks-api/issues/732)
 
+### Fixed
+- Contribution POST gives proper error message. [#714](https://github.com/rokwire/rokwire-building-blocks-api/issues/714)
```
Sorry, I did not see this early. This also needs to be moved to the unreleased section. Thank you.

ywkim312 replied:
changed

sandeep-ps replied:
Thanks.

sandeep-ps commented on contributions/catalog/webapps/templates/contribute/contribute.html:
```diff
@@ -787,7 +787,7 @@ <h3>Self-certification</h3>
         //success
         // window.location.href = "/contribute";
-        var x = document.getElementById('submit-pop').innerHTML = "The contribution form has been submitted!";
+        var x = document.getElementById('submit-pop').innerHTML = "The contribution form is being posted!";
```
```suggestion
        var x = document.getElementById('submit-pop').innerHTML = "Your Contribution is being submitted. Please wait!";
```

ywkim312 replied:
changed

sandeep-ps replied:
Thanks.
